src/libos/crates/host-socket/src/datagram/receiver.rs

```diff
@@ -65,11 +66,25 @@ impl<A: Addr, R: Runtime> Receiver<A, R> {
         self.common.pollee().del_events(Events::IN);
 
         // Copy data from the recv buffer to the bufs
-        let recv_bytes = inner.try_copy_buf(bufs);
-        if let Some(recv_bytes) = recv_bytes {
-            let recv_addr = inner.get_addr().unwrap();
-            self.do_recv(&mut inner);
-            return Ok((recv_bytes, recv_addr));
+        if A::domain() == Domain::Netlink {
+            let recv_bytes = inner.try_copy_buf_netlink(bufs);
+            if let Some(recv_bytes) = recv_bytes {
+                let recv_addr = inner.get_addr().unwrap();
+                if recv_bytes == 0 && inner.recv_len.is_none() {
```

Comment:
I think the `recv_bytes == 0 `condition is unnecessary. Because the `recv_len` is none , it means that the kernel buffer has been fully consumed.